iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkImpl.java

```diff
@@ -1262,7 +1262,8 @@ public ValidationLevel getValidationLevel() {
     public Network setMinimumAcceptableValidationLevel(ValidationLevel validationLevel) {
         Objects.requireNonNull(validationLevel);
         if (this.validationLevel == null) {
-            this.validationLevel = ValidationUtil.validate(Collections.unmodifiableCollection(index.getAll()), false, false, this.validationLevel, ReportNode.NO_OP);
+            ValidationLevel previous = this.minValidationLevel;
+            this.validationLevel = ValidationUtil.validate(Collections.unmodifiableCollection(index.getAll()), false, false, previous, ReportNode.NO_OP);
```

Comment:
I suggest to use `getValidationLevel()` here instead. It would highlight the intention:

```java
    @Override
    public Network setMinimumAcceptableValidationLevel(ValidationLevel minLevel) {
        Objects.requireNonNull(minLevel);
        ValidationLevel currentLevel = getValidationLevel();
        if (currentLevel.compareTo(minLevel) < 0) {
            throw new ValidationException(this, "Network should be corrected in order to correspond to validation level " + minLevel);
        }
        this.minValidationLevel = minLevel;
        return this;
    }
```

The variables were also renamed to increase readability.

Reply:
implemented your suggestion

Reply:
Thanks @zamarrenolm!